docs/swift-3-api.diff

```diff
@@ -0,0 +1,399 @@
++    public override func setValue(_ value: Any?, forUndefinedKey key: String)
+     public override class func shouldIncludeInDefaultSchema() -> Bool
+ public class ObjectUtil: NSObject
+ public final class ObjectSchema: CustomStringConvertible
+@@ -129,83 +141,85 @@
+ public final class Property: CustomStringConvertible
+     public var name: String 
+     public var type: PropertyType 
+-    public var indexed: Bool 
+-    public var optional: Bool 
++    public var isIndexed: Bool
++    public var isOptional: Bool
+     public var objectClassName: String? 
+     public var description: String 
+ public func == (lhs: Property, rhs: Property) -> Bool
+     public func == (lhs: Property, rhs: Property) -> Bool
+-public typealias NotificationBlock = (notification: Notification, realm: Realm) -> Void
+ public final class Realm
+     public var schema: Schema 
+     public var configuration: Configuration 
+     public var isEmpty: Bool 
+     public convenience init() throws
+     public convenience init(configuration: Configuration) throws
+-    public convenience init(fileURL: NSURL) throws
+-    public func write(@noescape block: (() throws -> Void)) throws
++    public convenience init(fileURL: URL) throws
++    public func write(block: () -> Void) throws
+     public func beginWrite()
+     public func commitWrite() throws
+     public func cancelWrite()
+-    public var inWriteTransaction: Bool
+-    public func add(object: Object, update: Bool = false)
+-    public func add<S: SequenceType where S.Generator.Element: Object>(objects: S, update: Bool = false)
+-    public func create<T: Object>(type: T.Type, value: AnyObject = [:], update: Bool = false) -> T
+-    public func dynamicCreate(className: String, value: AnyObject = [:], update: Bool = false) -> DynamicObject
+-    public func delete(object: Object)
+-    public func delete<S: SequenceType where S.Generator.Element: Object>(objects: S)
+-    public func delete<T: Object>(objects: List<T>)
+-    public func delete<T: Object>(objects: Results<T>)
+-    public func deleteAll()
+-    public func objects<T: Object>(type: T.Type) -> Results<T>
+-    public func dynamicObjects(className: String) -> Results<DynamicObject>
+-    public func objectForPrimaryKey<T: Object>(type: T.Type, key: AnyObject?) -> T?
+-    public func dynamicObjectForPrimaryKey(className: String, key: AnyObject?) -> DynamicObject?
+-    public func addNotificationBlock(block: NotificationBlock) -> NotificationToken
+-    public var autorefresh: Bool
++    public var isInWriteTransaction: Bool
++    public func add(_ object: Object, update: Bool = false)
++    public func add<S: Sequence>(_ objects: S, update: Bool = false) where S.Iterator.Element: Object
++    public func createObject<T: Object>(ofType type: T.Type, populatedWith value: Any = [:], update: Bool = false) -> T
++    public func createDynamicObject(ofType typeName: String, populatedWith value: Any = [:], update: Bool = false) -> DynamicObject
```

Comment:
This was named `dynamicCreate` because we specifically don't want people discovering it via autocomplete unless they're specifically looking for the dynamic object functionality.